From 8b8eefc074ed0219e565a43ba685f7b6917dca84 Mon Sep 17 00:00:00 2001
From: Yue Tao <Yue.Tao@windriver.com>
Date: Thu, 8 Aug 2019 02:38:45 -0700
Subject: [PATCH] ASoC: fsl_sai: change to TX/RX direction

The slave mode was changed to TX/RX direction by commit
5edb29fcee(ASoC: fsl_sai: handle slave mode per TX/RX direction).

The upstream commit ddb351145a9 still uses the is_slave_mode. To
change it to match the TX/RX mode.

Signed-off-by: Yue Tao <Yue.Tao@windriver.com>
---
 sound/soc/fsl/fsl_sai.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index 7443224b619d..6baae1c6196f 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -417,14 +417,14 @@ static int fsl_sai_set_dai_fmt_tr(struct snd_soc_dai *cpu_dai,
 	case SND_SOC_DAIFMT_CBS_CFS:
 		val_cr2 |= FSL_SAI_CR2_BCD_MSTR;
 		val_cr4 |= FSL_SAI_CR4_FSD_MSTR;
-		sai->is_slave_mode = false;
+		sai->slave_mode[tx] = true;
 		break;
 	case SND_SOC_DAIFMT_CBM_CFM:
 		sai->slave_mode[tx] = true;
 		break;
 	case SND_SOC_DAIFMT_CBS_CFM:
 		val_cr2 |= FSL_SAI_CR2_BCD_MSTR;
-		sai->is_slave_mode = false;
+		sai->slave_mode[tx] = true;
 		break;
 	case SND_SOC_DAIFMT_CBM_CFS:
 		val_cr4 |= FSL_SAI_CR4_FSD_MSTR;
-- 
2.17.1

